python/ray/data/_internal/execution/interfaces/op_runtime_metrics.py

```diff
@@ -123,6 +140,58 @@ def __init__(cls, name, bases, dict):
                 _METRICS.append(metric)
 
 
+def estimate_object_store_memory_per_node(
+    op: PhysicalOperator, state: "OpState"
+) -> Dict[str, int]:
+    usage_per_node = defaultdict(int)
+
+    # iterate through inqueue
+    for bundle in op._metrics._internal_inqueue._bundle_to_nodes:
+        for _, meta in bundle.blocks:
+            node_id = (
+                meta.exec_stats.node_id
+                if meta.exec_stats and meta.exec_stats.node_id
+                else _NODE_UNKNOWN
+            )
+            usage_per_node[node_id] += meta.size_bytes
```

Comment:
We count bundles in an operator's internal inqueue towards the previous operator's budget. So, if you're trying to reason about budgets and backpressure, counting these toward the current operator might be confusing.https://github.com/ray-project/ray/blob/333309f3b020a5ed073343e237f62f3cee698046/python/ray/data/_internal/execution/resource_manager.py#L108-L113

I think it might be more useful to mirror that logic here and exclude internal inqueues from an operator's estimate, but @raulchen curious to get your thoughts.

As an aside, I think it'd be good to have more explicit abstractions around bundle ownership (out of scope for this PR)